src/main/java/com/minh/project/controller/GameRestController.java

```diff
@@ -0,0 +1,124 @@
+package com.minh.project.controller;
+ 
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import com.minh.project.model.Game;
+import com.minh.project.service.GameService;
+ 
+@RestController
+public class GameRestController {
+	
+	Logger logger = LoggerFactory.getLogger(GameRestController.class);
+	
+    @Autowired
+    private GameService gameService;
+ 
+    
+    //-------------------Retrieve All Games--------------------------------------------------------
+    @RequestMapping(value = "/game/", method = RequestMethod.GET)
+    public ResponseEntity<?> listAllGames() {
+        List<Game> games = gameService.findAllGames();
+        if(games.isEmpty()){
+            return new ResponseEntity<List<Game>>(HttpStatus.NO_CONTENT);
+        }
+        return new ResponseEntity<List<Game>>(games, HttpStatus.OK);
+    }
+ 
+ 
+    
+    //-------------------Retrieve Single Game--------------------------------------------------------
+    @RequestMapping(value = "/game/{id}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
This is unsafe because someone can call this URL and get a game while bypassing the frontend validation. A way we can fix this is by putting the id the http.get call. Just something to keep in mind